contracts/src/controller/IL1EjectionController.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.13;
+
+/**
+ * @title IL1EjectionController
+ * @dev Interface for the L1 ejection controller that facilitates migrations of names
+ * between L1 and L2, as well as handling renewals. The controller is responsible
+ * for cross-chain communication.
+ */
+interface IL1EjectionController {
```

Comment:
In general, I don't think we can simply emit events and expect some indexing system to convert them into actions on an L2; instead we need to have a contract call the L1 side of a messaging bridge, which may use some kind of event system internally (but that's none of our concern). We can thus probably remove these events from the interface.